### DISCUSSION TITLE
Filter forks

### DISCUSSION THREAD

**Discussion by jeremy-rifkin** [community/community#15469] [Other Feature Feedback, Questions, & Ideas] [answered]
It'd be great to be able to filter forks. If I use `NOT repo:facebook/react-native` I still get flooded with a ton of forks. Something like `NOT repo:*/react-native` or `NOT is:fork` would be helpful.

**Comment by grappler** [comment]
Asked here too: https://github.com/github/feedback/discussions/13126

**Comment by brz95** [comment]
> Here's an example: https://cs.github.com/?scopeName=All+repos&scope=&q=%22curl_thread_t+Curl_thread_create%28unsigned+int+%28*func%29+%28void+*%29%2C+void+*arg%29%22
> 
> Consider these hits:
> 
> > nrfconnect/sdk-nrf
> > ‎ext/curl/lib/curl_threads.c‎
> > 1 match
> > C
> > main
> 
> ```
>   return 0;
> }
> 
> curl_thread_t Curl_thread_create(unsigned int (*func) (void *), void *arg)
> {
>   curl_thread_t t = malloc(sizeof(pthread_t));
>   struct Curl_actual_call *ac = malloc(sizeof(struct Curl_actual_call));
> ```
> 
> https://github.com/nrfconnect/sdk-nrf/blob/b6f76303347d8d5cf37c4186b4738cbce41b20c1/ext/curl/lib/curl_threads.c
> 
> It's a copy of [curl's](https://github.com/curl/curl/blob/5121d98c22dc077d72c68e7b9436a04f55bd7249/lib/curl_threads.c) -- not sure the specific version, and I'm not going to try.
> 
> > whitecatboard/Lua-RTOS-ESP32
> > ‎components/curl/lib/curl_threads.c‎
> > 1 match
> > C
> > master
> 
> ```
>   return 0;
> }
> 
> curl_thread_t Curl_thread_create(unsigned int (*func) (void *), void *arg)
> {
>   curl_thread_t t = malloc(sizeof(pthread_t));
>   struct curl_actual_call *ac = malloc(sizeof(struct curl_actual_call));
> ```
> 
> Ditto.
> 
> The point is that one doesn't want to see a committed copy of the code, just consumer code (which will be in the same repo, since, after all, they vendored the code, so presumably, they're making some calls to the library).

**Comment by brz95** [comment]
> Here's an example: https://cs.github.com/?scopeName=All+repos&scope=&q=%22curl_thread_t+Curl_thread_create%28unsigned+int+%28*func%29+%28void+*%29%2C+void+*arg%29%22
> 
> Consider these hits:
> 
> > nrfconnect/sdk-nrf
> > ‎ext/curl/lib/curl_threads.c‎
> > 1 match
> > C
> > main
> 
> ```
>   return 0;
> }
> 
> curl_thread_t Curl_thread_create(unsigned int (*func) (void *), void *arg)
> {
>   curl_thread_t t = malloc(sizeof(pthread_t));
>   struct Curl_actual_call *ac = malloc(sizeof(struct Curl_actual_call));
> ```
> 
> https://github.com/nrfconnect/sdk-nrf/blob/b6f76303347d8d5cf37c4186b4738cbce41b20c1/ext/curl/lib/curl_threads.c
> 
> It's a copy of [curl's](https://github.com/curl/curl/blob/5121d98c22dc077d72c68e7b9436a04f55bd7249/lib/curl_threads.c) -- not sure the specific version, and I'm not going to try.
> 
> > whitecatboard/Lua-RTOS-ESP32
> > ‎components/curl/lib/curl_threads.c‎
> > 1 match
> > C
> > master
> 
> ```
>   return 0;
> }
> 
> curl_thread_t Curl_thread_create(unsigned int (*func) (void *), void *arg)
> {
>   curl_thread_t t = malloc(sizeof(pthread_t));
>   struct curl_actual_call *ac = malloc(sizeof(struct curl_actual_call));
> ```
> 
> Ditto.
> 
> The point is that one doesn't want to see a committed copy of the code, just consumer code (which will be in the same repo, since, after all, they vendored the code, so presumably, they're making some calls to the library).



> But, even given your example, or rather, a [variant](https://cs.github.com/?q=CURLOPT_HEADERFUNCTION+NOT+%22curl%2C+CURLOPT_HEADERFUNCTION%22&p=4&pt=0721c5461381c6a2aea00d480eb8f128f9727a96fd9591fa475b23cdd37abfc99df298e15ee3296a2f59665dfc0ba6f70abb99e0267725a644ecaa375794495dca088156d6efa2fc80fea37f0416ea28fe7a8d94cd52b981f94ec32bb453b5bb63e6dfa411865745c7bd4c4234548a9c095e958f5bba4bdafcd1d1f06aba4cd22f845c06c149d38cf5260f18580329ba06f4bd0f1fc29adf5dfac9de9d6a3778b77b111a7eef24b27bcad56a98d37cb223a72b02e4aad047f8b7da19641efe004a0e7e83a34029ff2bfe43db92ce778d899408e017f0fcaaa40446919e31a0a366725fc3f1bfd8d80e696f759f983a56b0c56b7d24f38e0ae3a7ba040cf570af1f12150c34528e6cd8697261c51bd29f27ddff987c70be81f888c1dce116a5b6665133fac2bbfb303f2112d8085b3a2f57d2438269a75e315b19f896018bb887a3693eda7dc5c454b597becbd572f019feb4517f94b906d7d3520141be2f2f89c680534a258a2868bda3ccdd3f0bde2574e9bf5c092d578d8b404ba69ee1e6ce4609f4fd8eba5fd0f7938094494085ad&scope=&scopeName=All+repos)
> 
> Here's one where someone uploaded a copy of `curl` but didn't tell github it was a fork of curl/curl:
> 
> > [MikeRalphson/curl](https://cs.github.com/MikeRalphson/curl?q=CURLOPT_HEADERFUNCTION%20NOT%20%22curl%2C%20CURLOPT_HEADERFUNCTION%22)
> > [‎lib/url.c‎](https://cs.github.com/MikeRalphson/curl/blob/cba16bf38dc0fe40d65b5f4e266fbf96de014637/lib/url.c?q=CURLOPT_HEADERFUNCTION+NOT+%22curl%2C+CURLOPT_HEADERFUNCTION%22)
> > 1 match
> > C
> > master
> 
> ```c
>     if(!data->set.err)
>       data->set.err = stderr;
>     break;
>   case CURLOPT_HEADERFUNCTION:
>     /*
>      * Set header write callback
>      */
> ```
> 
> https://github.com/MikeRalphson/curl/blob/cba16bf38dc0fe40d65b5f4e266fbf96de014637/lib/url.c

**Comment by Wladefant** [comment]
This functionality is already on the [roadmap](https://github.com/orgs/community/discussions/13126#discussioncomment-2409411). 
And it seems that forks will be ignored by default in the future, but you can [add them](https://docs.github.com/en/search-github/searching-on-github/searching-in-forks) with fork:true, fork:only to the search.

**Comment by look** [comment, marked as answer]
Thanks for your patience, everyone. There is now an option to exclude forks (or limit your search to forks).

To exclude forks from your results:

```
NOT is:fork
```

To search only fork repositories:

```
is:fork
```

For more information, please [see the documentation](https://docs.github.com/en/search-github/github-code-search/understanding-github-code-search-syntax#is-qualifier).

**Reply by jeremy-rifkin** [reply]
How exciting, thank you!
